### PR TITLE
netrw: avoid clobbering filetype when opening netrw

### PR DESCRIPTION
**Pull request by wwade**
Some filetypes are set manually, or using autocmd, and do not work
properly with `filetype detect`, so in netrw, save and restore the
buffer's current filetype to be used in place of `filetype detect` in
`NetrwOptionRestore()`.

The original `filetype detect` call is left in place to account for the
possibility that `{a:vt}netrw_ftkeep` is not set.

### CODE REVIEW

**Comment by vim-ml** [comment]
On 2020-09-04, wwade wrote:
> Some filetypes are set manually, or using autocmd, and do not work
> properly with filetype detect, so in netrw, save and restore the
> buffer's current filetype to be used in place of filetype detect in
> NetrwOptionRestore().
> 
> The original filetype detect call is left in place to account for the
> possibility that {a:vt}netrw_ftkeep is not set.
> 
> -------------------------------------------------------------------------------
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/vim/vim/pull/6866
> 
> Commit Summary
> 
>   • netrw: avoid clobbering filetype when opening netrw
> 
> File Changes
> 
>   • M runtime/autoload/netrw.vim (7)
> 
> Patch Links:
> 
>   • https://github.com/vim/vim/pull/6866.patch
>   • https://github.com/vim/vim/pull/6866.diff

Vim plugins are maintained by their maintainers, usually the
authors, and not by other Vim developers.  Contact information
appears at the top of each plugin.

As it says at the top of the file containing ":help netrw", the
author of netrw is Charles Campbell.  Bug reports and enhancement
requests should be sent directly to him.

Regards,
Gary

**Comment by wwade** [comment]
Hi Gary,

Understood, but from what I can tell Chip is no longer maintaining this one, and Bram has actually taken over the work, at least according to the recent commits.  The last update on the vim.org/scripts page is from 2014, for example.

I tried reach out a couple of times, it's actually been a couple years with no response, so I'd given up on that route.

What do you think?
- Wade

PS apologies if there's a dupe message that's similar, but not identical, to this one.  I tried posting via google groups and I'm not entirely sure what happened.

**Comment by k-takata** [comment]
> but from what I can tell Chip is no longer maintaining this one,

He is still maintaining netrw. His mail address is written here:
https://github.com/vim/vim/blob/0ab2d326ff391b66f4f619b4ffd452cd369ee852/runtime/autoload/netrw.vim#L5
(Please remove "N", "O", "S", "P", "A", "M".)

>  The last update on the vim.org/scripts page is from 2014, for example.

The latest one is available here:
http://www.drchip.org/astronaut/vim/index.html#NETRW

**Comment by wwade** [comment]
Yes, I've emailed him on a couple of instances, but let me try again.

**Comment by k-takata** [comment]
Cc: @cecamp

**Comment by jamessan** [comment]
> Bram has actually taken over the work, at least according to the recent commits

Bram is the only one that commits to this repository, so don't go by that.

**Comment by wwade** [comment]
Thanks for the help, everyone, with the new email address, I managed to get in touch with Chip.  I'll close this one down for now.
